youtube_callback/sftp_id_retrieve.py

```diff
@@ -221,6 +227,95 @@ def domxml_parser(file):
+
+    with open(os.path.join(workdir, filename), 'rb') as csvfile:
+        file_reader = csv.reader(csvfile, delimiter=',')
+        headers = next(file_reader)
+        for column in headers:
+            if column == "Status":
+                status_index = headers.index(column)
+            elif column == "Video file":
+                file_suffix_index = headers.index(column)
+            elif column == "Video ID":
+                youtube_id_index = headers.index(column)
+
+        for row in file_reader:
+            video_url = row[file_suffix_index]
+            upload_data['status'] = row[status_index]
+            if upload_data['status'] == "Errors":
+                upload_data = _process_errors(upload_data, filename)
+
+            upload_data['youtube_id'] = row[youtube_id_index]
+
+            try:
+                upload_data['file_suffix'] = video_url.split("_")[1].split(".")[0]
+            except IndexError:
+                upload_data['file_suffix'] = 100
+
+    LOGGER.info('[YOUTUBE_CALLBACK] : Upload data : {upload_data}'.format(
+        upload_data=upload_data
+    ))
+
+    return upload_data
+
+
+def _process_errors(upload_data, reports_file):
+    """
+    :param upload_data : dict
+           reports_file : string
+    :return: upload_data : dict
+    """
+    errors_file = os.path.join(workdir, reports_file.replace("report-", "errors-"))
+
+    error_code_index = error_message_index = 0
+    error_message_pattern = re.compile('Duplicate video ID is \[(?P<thing>[0-9a-zA-Z_-]*)\]')
+
+    try:
+        with open(errors_file, 'rb') as csvfile:
+            file_reader = csv.reader(csvfile, delimiter=',')
+            headers = next(file_reader)
+            for column in headers:
+                if column == "Error code":
+                    error_code_index = headers.index(column)
+
+            for row in file_reader:
+                if row[error_code_index] == "VIDEO_REJECTED_DUPLICATE":
+                    upload_data['status'] = "Duplicate"
+                    error_message = row[error_message_index]
+                    youtube_id_search = error_message_pattern.search(error_message)
+                    if youtube_id_search:
+                        upload_data['youtube_id'] = youtube_id_search.groups()[0]
+                    else:
+                        LOGGER.error(
+                            '[YOUTUBE_CALLBACK] : Youtube callback returned Duplicate Video error but ' +
+                            'duplicate video ID could not be found. Upload data: {upload_data}. ' +
+                            'Error message: {error_message}'.format(
+                                upload_data=upload_data,
+                                error_message=error_message
+                            ))
+    except IOError:
+        LOGGER.error('[YOUTUBE_CALLBACK] : Could not open error file {file}'.format(
+            file=errors_file
+        ))
+        return upload_data
+    return upload_data
+
```

Comment:
👍 This is great!

Reply:
:(

Reply:
I meant it's great that you're implementing this loging! 🎉